### PR TITLE
 Catch foreign key violation exception on batch delete

### LINKED ISSUE #297: Catch ForeignKeyConstraintViolationException during batch action processing
There could be another context and/or request handler for [doctrine delete element](https://github.com/fsi-open/admin-bundle/blob/master/Doctrine/Admin/DeleteElement.php) which wraps deleting objects in transaction and `try`/`catch` which catches `ForeignKeyConstraintViolationException` and adds some error message about deleting of objects that are used somewhere. Unfortunately only the first used object could be printed in that message because after throwing an exception the `EntityManager` is closed and cannot be used for further entity processing.

### PR DESCRIPTION
**Pull request by szymach**
Resolves #297

### CODE REVIEW

File: spec/FSi/Bundle/AdminBundle/Admin/CRUD/Context/Request/DeleteRequestHandlerSpec.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    function it_catches_foreign_key_violation_exception(
+        FormEvent $event,
+        Request $request,
+        HandlerInterface $batchHandler
+    ) {
+        $batchHandler->handleRequest($event, $request)->willThrow(ForeignKeyConstraintViolationException::class);
+
+        $this->handleRequest($event, $request);
```
Comment (chives) [inline spec/FSi/Bundle/AdminBundle/Admin/CRUD/Context/Request/DeleteRequestHandlerSpec.php]: i think that this we should check if flash message is added

Comment (szymach) [reply spec/FSi/Bundle/AdminBundle/Admin/CRUD/Context/Request/DeleteRequestHandlerSpec.php]: I will after I get this to work, right now it is broken.

Comment (szymach) [reply spec/FSi/Bundle/AdminBundle/Admin/CRUD/Context/Request/DeleteRequestHandlerSpec.php]: The spec that is.

File: CHANGELOG-3.0.md [hunk 2]
```diff
@@ -193,3 +193,9 @@ utilize new php 7.1 features like scalar typehints, return typehints and nullabl
 All public methods of `FSi\Bundle\AdminBundle\Message\FlashMessages` now has more logical arguments order:
 `string $message, array $params = [], string $domain = 'FSiAdminBundle'` instead of previous
 `string $message, string $domain = 'FSiAdminBundle, array $params = []'`
+
+## Catching Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException on batch delete
+
+When attempting to delete entities with restricting relations, you would normally
+receive a SQL eror. Now the exception is caught and an error flash message is
```
Comment (chives) [inline CHANGELOG-3.0.md]: `error`

**Comment by szymach** [comment]
@fsi-open

**Pull request merged**